app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/ResolveCategoryAggregation.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Products\Query;
+
+use Magento\Catalog\Model\CategoryRepository;
+use Magento\Framework\Api\Search\AggregationValueInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Search\Response\Aggregation;
+use Magento\Framework\Search\Response\AggregationFactory;
+use Magento\Framework\Search\Response\Bucket;
+use Magento\Framework\Search\Response\BucketFactory;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Resolve category aggregation.
+ */
+class ResolveCategoryAggregation
+{
+    /**
+     * @var string
+     */
+    public const CATEGORY_BUCKET = 'category_bucket';
+
+    /**
+     * @var string
+     */
+    private const BUCKETS = 'buckets';
+
+    /**
+     * @var AggregationFactory
+     */
+    private $aggregationFactory;
+
+    /**
+     * @var BucketFactory
+     */
+    private $bucketFactory;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var CategoryRepository
+     */
+    private $categoryRepository;
+
+    /**
+     * @var array
+     */
+    private $resolvedChildrenIds = [];
+
+    /**
+     * @param AggregationFactory $aggregationFactory
+     * @param BucketFactory $bucketFactory
+     * @param StoreManagerInterface $storeManager
+     * @param CategoryRepository $categoryRepository
+     */
+    public function __construct(
+        AggregationFactory $aggregationFactory,
+        BucketFactory $bucketFactory,
+        StoreManagerInterface $storeManager,
+        CategoryRepository $categoryRepository
+    ) {
+        $this->aggregationFactory = $aggregationFactory;
+        $this->bucketFactory = $bucketFactory;
+        $this->storeManager = $storeManager;
+        $this->categoryRepository = $categoryRepository;
+    }
+
+    /**
+     * Get resolved category aggregation.
+     *
+     * @param array $categoryFilter
+     * @param Bucket[] $bucketList
+     *
+     * @return Aggregation
```

Comment:
Hello @korovitskyi 
Could you please merge tags to the common block?
Remove redundant line.

```suggestion
     * @param array $categoryFilter
     * @param Bucket[] $bucketList
     * @return Aggregation
```